src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -0,0 +1,411 @@
+//! Client library for the JSON-RPC-based File Manager service.
+
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+#![warn(unsafe_code)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+
+use crate::prelude::*;
+
+use crate::common::UTCDateTime;
+
+use json_rpc::api::Result;
+use json_rpc::Handler;
+use json_rpc::make_arg;
+use json_rpc::make_param_map;
+use json_rpc::make_rpc_methods;
+use futures::Stream;
+use serde::Serialize;
+use serde::Deserialize;
+use std::future::Future;
+use uuid::Uuid;
+
+
+
+// =============
+// === Event ===
+// =============
+
+/// Event emitted by the File Manager `Client`.
+pub type Event = json_rpc::handler::Event<Notification>;
+
+
+
+// ============
+// === Path ===
+// ============
+
+/// Path to a file.
+#[derive(Clone,Debug,Display,Eq,Hash,PartialEq,PartialOrd,Ord)]
+#[derive(Serialize, Deserialize)]
+#[derive(Shrinkwrap)]
+pub struct Path(pub String);
+
+impl Path {
+    /// Wraps a `String`-like entity into a new `Path`.
+    pub fn new(s:impl Str) -> Path {
+        Path(s.into())
+    }
```

Comment:
This should be `From` instance

Reply:
Same as the comment above. Path is already being reworked

Reply:
Same as the comment above then.